qemu/tests/cgroup.py

```diff
@@ -2059,6 +2061,51 @@ def memory_move():
         else:
             return (out)
 
+    def cfs_bandwidth():
+        """
+        CFS bandwidth limit
+        """
+        def stress_thread():
+            session.cmd("stress -c 1 -t 100", 120)
+
+        logging.info("Setup test")
+        modules = CgroupModules()
+        if (modules.init(['cpu']) != 1):
+            raise exceptions.TestFail("Can't mount cpu cgroup modules")
+        cgroup = Cgroup('cpu', '')
+        cgroup.initialize(modules)
+
+        # Create VM with smp=1
+        params['smp'] = 1
+        params["vcpu_sockets"] = 1
+        vm_name = params["main_vm"]
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(params["main_vm"])
+        login_timeout = int(params.get("login_timeout", 360))
+        session = vm.wait_for_login(timeout=login_timeout)
```

Comment:
You don't use this session in the main thread, better open up and close it where use it.

Reply:
I want to ensure run stress and top at the same time, so I only put session.cmd in stress_thread(). If I put session in stress_thread(), it may need time to login session. I will close it at the end.

Reply:
@YanhuiMa Well, if so, you need to at least check if stress is installed, we can hardly presume that the application is pre-installed.